### ISSUE TITLE
DOC: optimize: linprog method-specific function signature

### ISSUE THREAD

**Issue by mdhaber** [scipy/scipy#12264]
The `linprog` method-specific function signature has been incorrect since 1.2.0. 

In [1.1.0](https://docs.scipy.org/doc/scipy-1.1.0/reference/optimize.linprog-interior-point.html), documentation for `method='interior-point'` (for example) was correct; it looked like:
![image](https://user-images.githubusercontent.com/6570539/83223114-f76e7400-a12e-11ea-85e2-6b27f8109688.png)

Since [1.2.0](https://docs.scipy.org/doc/scipy-1.2.0/reference/optimize.linprog-interior-point.html), it's been more like:
![image](https://user-images.githubusercontent.com/6570539/83223200-2a186c80-a12f-11ea-8a9e-a8c055475383.png)

The cause seems to be the change in the signature of the method-specific function that `linprog` dispatches to, `_linprog_ip`. The function [used to](https://github.com/scipy/scipy/blob/14142ff70d84a6ce74044a27919c850e893648f7/scipy/optimize/_linprog_ip.py#L1747) accept most of the same arguments as `linprog`, [now](https://github.com/scipy/scipy/blob/0b4a8a293f4fe16e1ba2537f7eaccfa8d09ff9a5/scipy/optimize/_linprog_ip.py#L822) the signature is quite different. Even though `_linprog_ip`'s signature has changed, though, users should still be calling `linprog` with `method='interior-point'` as the documentation for 1.1.0 suggests.

The problem is not specific to `method='interior-point'`; the same sort of thing is true of `'simplex'`, `'revised-simplex'`, and the new [HiGHS methods](https://github.com/scipy/scipy/pull/12043).

What is the best way to correct this? 

@rgommers @mckib2

**Comment by mdhaber** [comment]
This caused gh-12637.

If you know more about Sphinx than me, I'd appreciate a lead. Thanks!

**Comment by peterbell10** [comment]
I think the `scipyoptdoc` is filtering out the arguments because it thinks they aren't used by the underlying method:
https://github.com/scipy/scipy/blob/c1372d8aa90a73d8a52f135529293ff4edb98fc8/doc/source/scipyoptdoc.py#L116-L119

Perhaps the list of exceptions should be made into a parameter in the `.rst` file.

**Comment by mdhaber** [comment]
Thanks @peterbell10!

That seems to be part of it. I added `A_ub`, `b_ub`, `A_eq`, `b_eq`, and `bounds` into that list and now I see:

![image](https://user-images.githubusercontent.com/6570539/89233990-bf0c6b80-d59f-11ea-9c34-76348d1d828b.png)

So now those show up in the signature as parameters, as desired. However, we still have the following problems:

- `c0`, `A`, `b`, or `postsolve_args` appear as options in the signature at the top,
- `c0`, `A`, `b`, or `postsolve_args` appear as parameters below
- `A_ub`, `b_ub`, `A_eq`, `b_eq`, and `bounds` don't appear as parameters below, but they should

Basically, what we want is for the method-specific documentation to look exactly like the top level `linprog` documentation, except we want `method="method-name" to appear in the signature and the method-specific options and notes to appear below. Thoughts on how we could get tha behaviort?

**Comment by peterbell10** [comment]
The content below the signature comes from the docstring for `_linprog_ip`. Some possible fixes:
- change the parameter documentation for `_linprog_ip` (won't fix signature)
- point `scipyoptdoc` at an unused function that has the signature and docstring you want
- or possibly even forego `scipyoptdoc` entirely for `linprog`

**Comment by mdhaber** [comment]
I'm going with 'point `scipyoptdoc` at an unused function that has the signature and docstring you want`

Thanks!
